test/smoke/dashCore.js

```diff
@@ -1,18 +1,115 @@
+
+
+describe('All nodes', () => {
+  let nodeNames = networkConfig.inventory.masternodes.hosts;
+  nodeNames = nodeNames.concat(networkConfig.inventory['wallet-nodes'].hosts);
+  // TODO miner rpc call stuck. why?
+  // nodeNames = nodeNames.concat(networkConfig.inventory['miners'].hosts);
+  nodeNames = nodeNames.concat(networkConfig.inventory['full-nodes'].hosts);
+  nodeNames.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      it('should have correct network type', async () => {
+        const networkInfo = await rpc.getNetworkInfo();
+        expect(networkInfo.error).to.be.equal(null);
+        expect(networkInfo.result.networkactive).to.be.equal(true);
+        expect(networkInfo.result.subversion).to.have.string(`(${networkConfig.network.type}=${networkConfig.network.name})/`);
+      });
+    });
   });
-  describe('Masternodes', () => {
-    // masternode status
-    it('should be masternodes');
+
+  // The number of blocks should be almost the same (-3/+3) and block hash of particular
+  // block height should be the same. Using `blocks`
+  // and `bestblockhash` from `GetBlockChainInfo`.
+  // Verify it on masternodes, wallet-nodes, full-nodes, miners
+  it('should have correct blockhash and blocks count', async () => {
+    const blockCounts = [];
+    const blockhashes = {};
+    let blockchainInfo;
+    for (let i = 0; i < nodeNames.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeNames[i]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      const blockCount = await rpc.getBlockCount();
+      blockchainInfo = await rpc.getBlockchainInfo();
+      blockCounts.push(blockCount.result);
+      if (!blockhashes[blockchainInfo.result.blocks]) {
+        blockhashes[blockchainInfo.result.blocks] = blockchainInfo.result.bestblockhash;
+      }
+    }
+    expect(blockhashes[blockchainInfo.result.blocks]).to.be.equal(
+      blockchainInfo.result.bestblockhash,
+    );
+    expect(Math.max(...blockCounts) - Math.min(...blockCounts)).to.be.below(3);
   });
+});
+
+describe('Masternodes', () => {
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      // masternode status
+      it('should masternodes be enabled', async () => {
+        const masternodelist = await rpc.masternodelist();
```

Comment:
Why not just https://dash-docs.github.io/en/developer-reference#masternode--noref---status?

Reply:
rpc masternodelist returns list of all nodes and their statuses. so, currently we verify that rpc to every node returns status "enabled" by all nodes. if you suggest to use rpc.masternode('status') we get status only by one node.
what is  the better option?

Reply:
Yes, probably `masternode list` check more bulletproof :)